### PR TITLE
Feature/terminate contract negotiation

### PR DESCRIPTION
**Pull request by IgorBalog-Eng**
### Changed

 - ContractNegotiation.callbackAddress on consumer side value set to provider address (protocol address)
 - Terminate negotiation API request - based on role, it sends to provider protocol address or consumer callback address

### Added

 - Provider handleTermination request
 - New integration test to cover contract negotiation termination logic (without change of initial_data.json)

### CODE REVIEW

File: negotiation/src/main/java/it/eng/negotiation/rest/protocol/ProviderContractNegotiationController.java [hunk 1]
```diff
@@ -111,11 +112,14 @@ public ResponseEntity<JsonNode> handleVerifyAgreement(@PathVariable String provi
...
+        ContractNegotiationTerminationMessage contractNegotiationTerminationMessage = 
         		Serializer.deserializeProtocol(contractNegotiationTerminationMessageJsonNode, ContractNegotiationTerminationMessage.class);
 
-        ContractNegotiationErrorMessage error = methodNotYetImplemented();
-        return ResponseEntity.internalServerError().contentType(MediaType.APPLICATION_JSON).body(Serializer.serializeProtocolJsonNode(error));
+        providerService.handleTerminationRequest(providerPid, contractNegotiationTerminationMessage);
+        
+        return ResponseEntity.ok()
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON.toString())
+                .build();
```
Comment (davidjovanovic) [inline negotiation/src/main/java/it/eng/negotiation/rest/protocol/ProviderContractNegotiationController.java:122]: Why is this changed and not .contentType(MediaType.APPLICATION_JSON)?

**Review by davidjovanovic — APPROVED** [review]

**Review by marest94 — APPROVED** [review]
